### PR TITLE
Integration tests improvements

### LINKED ISSUE #411: Integration tests improvements
- [x] Improve logging
- [x] Add test data for documents and arrays
- [x] Add custom `assertEqual` with NaN handling

### PR DESCRIPTION
**Pull request by AlekSi**
Add test data for documents and arrays.
Fix NaNs comparision.

Closes #411.

### CODE REVIEW

File: integration/helpers.go [hunk 1]
```diff
@@ -20,8 +20,90 @@ import (
 
 	"github.com/stretchr/testify/require"
 	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/testutil"
 )
 
+// convert converts given driver value (bson.D, bson.A, etc) to FerretDB types package value.
```
Comment (seeforschauer) [inline integration/helpers.go:29]: :+1:

File: integration/helpers.go [hunk 2]
```diff
@@ -20,8 +20,90 @@ import (
...
+		}
+	case int32:
+		return v
+	case primitive.Timestamp:
+		return types.Timestamp(uint64(v.I)<<32 + uint64(v.T))
+	case int64:
+		return v
+	default:
+		t.Fatalf("unexpected type %T", v)
+		panic("not reached")
```
Comment (seeforschauer) [inline integration/helpers.go:79]: test fatal should fail, isn't it? it's not fails?

File: integration/query_comparision_test.go [hunk 3]
```diff
@@ -15,25 +15,43 @@
...
+			t.Parallel()
+
+			var actual []bson.D
+			cursor, err := collection.Find(ctx, bson.D{{"_id", bson.D{{"$eq", id}}}}, opts)
+			require.NoError(t, err)
+
+			err = cursor.All(ctx, &actual)
+			require.NoError(t, err)
+			require.Len(t, actual, 1)
+			assertEqualDocuments(t, expected, actual[0])
```
Comment (seeforschauer) [inline integration/query_comparision_test.go:53]: actual[0] - the first only? then should the limit be added to the Find query options? it won't fail if there are will be more than 1 elements in actual.

File: integration/helpers.go (hunk 2 above)
Comment (AlekSi) [reply integration/helpers.go:79]: It does fail, but compile is not smart enough to understand that return is not needed there
```
./helpers.go:81: missing return
```

I wanted to be sure that all cases have returns, so panic acts there as impossible [terminal statement](https://go.dev/ref/spec#Terminating_statements)

File: integration/query_comparision_test.go (hunk 3 above)
Comment (AlekSi) [reply integration/query_comparision_test.go:53]: The previous line ensures that there is exactly one element

**Review by seeforschauer — APPROVED** [review]

**Pull request merged**